app/src/main/java/com/sample/appconnectsample/LoginActivity.java

```diff
@@ -53,6 +53,22 @@ public void doLogInButton(View source) {
         new LogInTask().execute(username, password);
     }
 
+    public void doRegisterButton(View source) {
+        Intent intent = new Intent(LoginActivity.this, RegistrationEmailActivity.class);
+        startActivityForResult(intent, RegistrationEmailActivity.REGISTRATION_REQUEST, null);
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        if (requestCode == RegistrationEmailActivity.REGISTRATION_REQUEST) {
+            if (resultCode == RESULT_OK) {
+                // The user registered successfully, set the email field, and clear password
+                usernameField.setText(data.getStringExtra("email"));
+                passwordField.setText("");
+            }
+        }
```

Comment:
Nitpick: nested ifs is probably overkill for two conditions. Could combine them or return early.